lib/MongoDB/_Topology.pm

```diff
@@ -597,8 +597,9 @@ sub _check_wire_versions {
         my ( $server_min_wire_version, $server_max_wire_version ) =
           @{ $server->is_master }{qw/minWireVersion maxWireVersion/};
 
-        $server_max_wire_version = 0 unless defined $server_max_wire_version;
-        $server_min_wire_version = 0 unless defined $server_min_wire_version;
+        # set to -1 as could be undefined, or 0.
```

Comment:
This should stay "0" because missing is equivalent to wire version "0" and we don't need it to force the compatibility logic to fail, as the Perl driver is special and has to deviate from the spec and support older versions (because of some large customers that asked for legacy support).